compiler/rustc_data_structures/src/stable_hasher.rs

```diff
@@ -209,6 +209,12 @@ impl_stable_hash_via_hash!(i128);
 impl_stable_hash_via_hash!(char);
 impl_stable_hash_via_hash!(());
 
+impl<CTX> HashStable<CTX> for ! {
+    fn hash_stable(&self, _ctx: &mut CTX, _hasher: &mut StableHasher) {
+        unreachable!()
```

Comment:
`match *self {}`?

Reply:
Looks like standard library impls for `!` use just `*self`.